docs/index.md

```diff
@@ -3,85 +3,326 @@ hide:
+  <div class="section-chip blue">Modern UI principles</div>
+  <div class="section-headline">
+    Declarative UI that's easy to understand
+  </div>
+  <div class="section-body">
+    Mesop embraces a declarative approach to UI development, making your code more intuitive and maintainable. Say goodbye to complex imperative logic and hello to clean, expressive code. Build user interaces using battle-tested patterns, without leaving Python.
+  </div>
+</section>
+
+```py
+@me.stateclass
+class State:
+    image_data: str
+    detections: list[Detection]
 
-```sh
-pip install mesop
-```
 
-__Step 2:__ Copy the example above into `main.py`
+@me.page()
+def object_detector():
+    state = me.state(State)
 
-__Step 3:__ Run the app
+    me.text("Real-time Object Detection", type="headline-4")
+    me.uploader(label="Upload an image", on_upload=on_image_upload)
 
-```sh
-mesop main.py
+    if state.image_data:
+        me.image(src=f"data:image/jpeg;base64,{state.image_data}")
+
+    if state.detections:
+        me.text("Detected Objects:", type="headline-5")
+        for detection in state.detections:
+            detection_component(detection)
+
+def detection_component(detection):
+    me.text(f"{detection.obj}: {detection.confidence:.2f}")
+
+def on_image_upload(e: me.UploadEvent):
+    state = me.state(State)
+    state.image_data = base64.b64encode(e.file.read()).decode()
+    state.detections = detect_objects(e.file)
 ```
 
-## Next Steps
+<section class="container pink-gradient-background">
+  <div class="section-chip pink">Building blocks</div>
+  <div class="section-headline">
+    Ready-to-use components
+  </div>
+  <div class="section-body">
+    Mesop provides a versatile range of <a href="./components/">30 components</a>, from low-level building blocks to high-level, AI-focused components. This flexibility lets you rapidly prototype ML apps or build custom UIs, all within a single framework that adapts to your project's use case.
+
+    <a class="section-button" href="./components/chat">
+      AI components
+      <div class="section-button-subtext">e.g. chat, text to image</div>
+    </a>
+    <a class="section-button" href="./components/input">
+      Form components
+      <div class="section-button-subtext">e,g. input, checkbox, radio</div>
+    </a>
+    <a class="section-button" href="./components/table">
+      Data display components
+      <div class="section-button-subtext">e,g. table, plot</div>
+    </a>
+  </div>
+</section>
+
+<section class="container teal-gradient-background">
+  <div class="section-chip teal">Build anything</div>
+  <div class="section-headline">
+    Build any user interface you can imagine
+  </div>
+  <div class="section-body">
+With Mesop, you can build virtually any web-based user interface or application you can imagine. From quick prototypes to enterprise tools, Mesop provides the customizability to bring your ideas to life.
+  </div>
+  <div class="center">
```

Comment:
I'm not a fan of the centering of the iframe against the left alignment of the content. But that is more of a pet peeve of mine. So I think fine as is? How come the iframe doesn't just fit the full width of the section?